src/js/components/Playlists/Playlists.react.js

```diff
@@ -26,45 +27,36 @@ export default class Playlists extends Component {
         this.state = {};
     }
 
-    render() {
-
-        let content;
+    createPlaylist() {
+        AppActions.playlists.create('New playlist', true);
+    }
 
-        // TODO (y.solovyov): move all these cases to separate components
+    getContent() {
         if(this.props.playlists === null) {
-            content = (
-                <div className='full-message'>
-                    <p>Loading playlists</p>
-                </div>
-            );
-        } else if(this.props.playlists.length === 0) {
-            content = (
-                <div className='full-message'>
-                    <p>You haven't created any playlist yet</p>
-                    <p className='sub-message'>
-                        <a onClick={ () => {
-                            AppActions.playlists.create('New playlist', true);
-                        } }>
-                            create one now
-                        </a>
-                    </p>
-                </div>
-            );
-        } else if(!this.props.params.playlistId) {
-            content = (
-                <div className='full-message'>
-                    <p>Select a playlist in the menu on the left</p>
-                </div>
+            return <FullViewMessage message='Loading playlists' />;
+        }
+
+        if(this.props.playlists.length === 0) {
+            return (
+                <FullViewMessage message="You haven't created any playlist yet">
+                    <a onClick={ this.createPlaylist }> create one now </a>
+                </FullViewMessage>
             );
-        } else {
-            content = React.cloneElement(this.props.children, { ...this.props });
         }
 
+        if(!this.props.params.playlistId) {
+            return <FullViewMessage message='Select a playlist in the menu on the left' />;
+        }
+
+        return React.cloneElement(this.props.children, { ...this.props });
+    }
+
+    render() {
         return (
             <div className='view view-playlists'>
                 <PlaylistsNav playlists={ this.props.playlists } />
                 <div className='playlist'>
-                    { content }
```

Comment:
This is perfectly fine. I want to keep all render logic inside the `render` method for now